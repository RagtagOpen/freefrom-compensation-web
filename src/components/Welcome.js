import React, { Component } from "react"

import { connect } from "react-redux"

const mapStateToProps = ({ hasAcknowledged }) => {
  return { hasAcknowledged }
}

const mapDispatchToProps = dispatch => {
  return {
    acknowledge: bool => dispatch({ type: `UPDATE ACKNOWLEDGEMENT` }),
  }
}

class Welcome extends Component {
  state = {}
  render() {
    return (
      <div style={{ margin: "1rem", padding: "1rem" }}>
        <h2>Welcome to the new FreeFrom Quiz.</h2>
        <div>
          <label htmlFor="Acknowledgement checkbox">
            <input type="checkbox" onChange={this.props.acknowledge} />
          </label>
          <span>
            By checking this box, you acknowledge that you have read and agree
            to the terms and conditions provided.
          </span>
        </div>
      </div>
    )
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Welcome)
