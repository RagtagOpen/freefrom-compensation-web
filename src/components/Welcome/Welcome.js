import React, { Component } from "react"
import { connect } from "react-redux"
import Typography from '@material-ui/core/Typography';

const mapStateToProps = ({ hasAcknowledged }) => {
  return { hasAcknowledged }
}

const mapDispatchToProps = dispatch => {
  return {
    acknowledge: bool => dispatch({ type: `UPDATE ACKNOWLEDGEMENT` }),
  }
}

export class Welcome extends Component {
  state = {}
  render() {
    return (
      <div style={{ margin: "1rem", padding: "1rem" }}>
        <Typography variant="h3">What is Your Compensation Type?</Typography>
        <div>
          <label htmlFor="Acknowledgement checkbox">
            <input type="checkbox" onChange={this.props.acknowledge} />
          </label>
          <span>
            By checking this box, you acknowledge that you have read and agree to the terms and conditions provided.
          </span>
        </div>
      </div>
    )
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Welcome)
