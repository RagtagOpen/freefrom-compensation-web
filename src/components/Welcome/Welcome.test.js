import React from "react"
import renderer from "react-test-renderer"

import { Welcome } from "./Welcome"

describe("Welcome", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Welcome hasAcknowledged="true" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})