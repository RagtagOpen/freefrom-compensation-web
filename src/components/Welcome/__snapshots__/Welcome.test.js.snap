// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome renders correctly 1`] = `
<div
  style={
    Object {
      "margin": "1rem",
      "padding": "1rem",
    }
  }
>
  <h2>
    Welcome to the new FreeFrom Quiz.
  </h2>
  <div>
    <label
      htmlFor="Acknowledgement checkbox"
    >
      <input
        type="checkbox"
      />
    </label>
    <span>
      By checking this box, you acknowledge that you have read and agree to the terms and conditions provided.
    </span>
  </div>
</div>
`;
